examples/apps/color_finder.py

```diff
@@ -0,0 +1,659 @@
+            d = color_distance_sqr((r, g, b), color_values)
+            if d < min_distance:
+                min_distance = d
+                closest_color = color_name
+        return closest_color
+
+    def fill_in_gaps_of_pixel_matrix(self):
+        ''' Fills in squares in self.pixel_matrix that meet the condition
+            in self.pixel_is_surrounded_by_color.
+        '''
+        for i in range(self.pixel_matrix.num_cols):
+            for j in range(self.pixel_matrix.num_rows):
+                color = self.pixel_matrix.surrounded(i,j)
+                if color:
+                    self.pixel_matrix.at(i, j).set(color)
+
+    def get_low_res_view(self):
+        ''' Returns a low-resolution version of Cozmo's camera view.'''
+        image = gray_world(self.robot.world.latest_image.raw_image)
+        return image.resize((self.downsize_width, self.downsize_height), resample = Image.LANCZOS)
+
+    def on_finding_a_blob(self, blob_info):
+        ''' Determines whether Cozmo should continue to look at the blob, or drive towards it.
+            
+            Args:
+                blob_info (int, int): coordinates of the blob's center in self.pixel_matrix
+        '''
+        self.robot.set_center_backpack_lights(map_color_to_light[self.color_to_find])
+
+        x, y = blob_info
+        WM = 10.0*32/self.downsize_width  # WM and HM are multipliers that adjust the amount to turn
+        HM = 10.0*24/self.downsize_height # based on the current dimensions of self.pixel_matrix.
+        # 'fov' stands for 'field of view'. This is the angle amount
+        # that Cozmo can see to the edges of his camera view.
+        amount_move_head = radians(self.fov_y.radians*((CAMERA_HEIGHT/2)-y*HM)/CAMERA_HEIGHT)
+        amount_rotate = radians(self.fov_x.radians*((CAMERA_WIDTH/2)-x*WM)/CAMERA_WIDTH)
+        if self.moved_too_far_from_center(amount_move_head, amount_rotate):
+            self.state = FOUND_COLOR_STATE
+        if self.state != DRIVING_STATE:
+            self.look_at_it(amount_move_head, amount_rotate)
+        else:
+            self.drive_at_it()
+
+    def moved_too_far_from_center(self, amount_move_head, amount_rotate):
+        ''' Decides whether the center of the blob is too far from the center of Cozmo's view.
+
+            Args:
+                amount_move_head (cozmo.util.Angle): 
+                    the perceived vertical distance of the blob from center-screen
+                amount_rotate (cozmo.util.Angle): 
+                    the perceived horizontal distance of the blob from center-screen
+
+            Returns:
+                too_far (bool): whether the object is too far from center-screen
+        '''
+        too_far_vertical = (amount_move_head.abs_val > self.fov_y/8)
+        too_far_horizontal = (amount_rotate.abs_val > self.fov_x/8)
+        too_far = too_far_vertical or too_far_horizontal
+        return too_far
+
+    def look_at_it(self, amount_move_head, amount_rotate):
```

Comment:
look_at_it could be renamed to something more descriptive like look_at_color_blob or similar